src/main/java/com/avispl/symphony/dal/avdevices/audionetworkinterface/qsc/undnemo/QSCUndnemoCommunicator.java

```diff
@@ -0,0 +1,805 @@
+		stats.put(String.format("%s#%s", oldActiveIndexGroupName, QSCUndnemoMetric.CHANNEL_INFO_ENABLE_STATE.getName()), oldActiveIndexEnableState);
+		stats.put(String.format("%s#%s", oldActiveIndexGroupName, QSCUndnemoMetric.CHANNEL_INFO_DEVICE_NAME.getName()), oldActiveIndexDeviceName);
+		stats.put(String.format("%s#%s", oldActiveIndexGroupName, QSCUndnemoMetric.CHANNEL_INFO_CHANNEL_NAME.getName()), oldActiveIndexChannelName);
+		stats.put(String.format("%s#%s", oldActiveIndexGroupName, QSCUndnemoMetric.CHANNEL_INFO_DISPLAY_NAME.getName()), oldActiveIndexDisplayName);
+
+		localExtendedStatistics.setStatistics(stats);
+		localExtendedStatistics.setControllableProperties(controllableProperties);
+	}
+
+	/**
+	 * Handle control active channel index with filter option. This method is used to handle every statistics locally -
+	 * to save the number of requests made to the device.
+	 *
+	 * @param value value of new active channel
+	 * @param indexList list of filter indexes
+	 * @param currentActiveChannelIndex current active channel index
+	 * @throws Exception if fail to get UDP response
+	 */
+	private void handleActiveChannelControlWithFilter(String value, List<Integer> indexList, String currentActiveChannelIndex) throws Exception {
+		String response = getUDPResponse(QSCUndnemoUDPCommand.SET_ACT_CH_IDX.getCommand() + QSCUndnemoConstant.SPACE + value);
+		if (QSCUndnemoConstant.NACK.equals(response)) {
+			throw new CommandFailureException(this.getAddress(), QSCUndnemoUDPCommand.SET_ACT_CH_IDX.getCommand(), String.format("Fail to set active channel index with value: %s", value));
+		}
+		Map<String, String> stats = localExtendedStatistics.getStatistics();
+		List<AdvancedControllableProperty> controls = localExtendedStatistics.getControllableProperties();
+		synchronized (channelInfoList) {
+			for (ChannelInfo channelInfo : channelInfoList) {
+				String channelInfoIndex = channelInfo.getChannelInfoIndex();
+				if (channelInfoIndex.equals(currentActiveChannelIndex) && !indexList.contains(Integer.valueOf(channelInfoIndex))) {
+					channelInfoList.remove(channelInfo);
+					break;
+				}
+			}
+		}
+		if (indexList.contains(Integer.valueOf(value))) {
+			populateChannelInfoMonitoringAndControllingProperties(stats, controls);
+		} else {
+			List<Integer> listOfChannelIndex = new ArrayList<>();
+			listOfChannelIndex.add(Integer.valueOf(value));
+			retrieveChannelInfo(listOfChannelIndex);
+			populateChannelInfoMonitoringAndControllingProperties(stats, controls);
+		}
+		localExtendedStatistics.setStatistics(stats);
+		localExtendedStatistics.setControllableProperties(controls);
+	}
+
+
+	/**
+	 * Prepare list of index before submitting.
+	 *
+	 * @param x begin index
+	 * @param y end index
+	 */
+	private void submitThread(int x, int y) {
+		List<Integer> listIndexes = new ArrayList<>();
+		for (int i = x; i <= y; i++) {
+			listIndexes.add(i);
+		}
+		executorService.submit(channelDataLoader = new QSCChannelDataLoader(listIndexes));
```

Comment:
The assignments should not be made from within sub-expressions, it hard to spot and therefore make the code less readable. Please fix it

Reply:
Updated.